api/clients/eigenda_client_v2.go

```diff
@@ -0,0 +1,142 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/api/clients/codecs"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/hashicorp/go-multierror"
+	"math/rand"
+)
+
+// EigenDAClientV2 provides the ability to get blobs from the relay subsystem, and to send new blobs to the disperser.
+type EigenDAClientV2 interface {
```

Comment:
I'm going to start a crusade in our repo against interfaces defined generically like this. Don't think this should live here. Interfaces (in 95% of cases) are meant to be defined at each respective callsite (because different callsites might need very different subsets of methods, only which they need to mock for their tests).